crypt/src/crypt.cpp

```diff
@@ -106,6 +106,72 @@ static int Crypt_Base64Decode(lua_State* L)
     return 1;
 }
 
+static int Crypt_EncryptXTEA(lua_State* L)
+{
+    DM_LUA_STACK_CHECK(L, 1);
+
+    size_t srclen;
+    const char* src = luaL_checklstring(L, 1, &srclen);
+
+    size_t keylen;
+    const char* key = luaL_checklstring(L, 2, &keylen);
+    if (keylen > 16)
+    {
+      luaL_error(L, "key should be <=16");
+    }
+
+    uint32_t dstlen = srclen;
+    char* dst = (char*) malloc(dstlen);
+    memcpy(dst, src, dstlen);
+    dmCrypt::Result result = dmCrypt::Encrypt(dmCrypt::ALGORITHM_XTEA, (uint8_t*)dst, dstlen, (uint8_t*)key, keylen);
+    if (result != dmCrypt::RESULT_OK)
+    {
+        free(dst);
+        luaL_error(L, "error when encrypt");
```

Comment:
Note:
From [Lua Manual](https://www.lua.org/manual/5.1/manual.html#luaL_error): "This function never returns, but it is an idiom to use it in C functions as return luaL_error(args)."

Reply:
Also, since you are using our `DM_LUA_STACK_CHECK`, you must match it with `DM_LUA_ERROR`
See [documentation](https://defold.com/ref/stable/dmScript/?q=dm_lua_stack_check#DM_LUA_ERROR)

Reply:
Thanks. I fixed it. Now return DM_LUA_ERROR